jena-fuseki2/jena-fuseki-main/src/main/java/org/apache/jena/fuseki/main/FusekiServer.java

```diff
@@ -736,6 +738,41 @@ private void serviceEndpointOperation(String datasetName, String endpointName, O
             dap.getDataService().addEndpoint(endpoint);
         }
 
+        /**
+         * Set the number threads used by Jetty. This uses a {@code org.eclipse.jetty.util.thread.QueuedThreadPool} provided by Jetty.
+         * <p>
+         * Argument order is (minThreads, maxThreads).
+         * <p>
+         * <ul>
+         * <li>Use (-1,-1) for Jetty "default". The Jetty 9.4 defaults are (min=8,max=200).
+         * <li>If (min != -1, max is -1) then the default max is 20.
+         * <li>If (min is -1, max != -1) then the default min is 2.
+         * </ul>
+         */
+        public Builder numServerThreads(int minThreads, int maxThreads) {
+            if ( minThreads >= 0 && maxThreads > 0 ) {
+                if ( minThreads > maxThreads )
+                    throw new FusekiConfigException(String.format("Bad thread setting: (min=%d, max=%d)", minThreads, maxThreads));
+            }
```

Comment:
Just wondering if this is set by a user external configuration (e.g. assembler file). In such case, couldn't the user pass `-1, -11` by accident? Does that create a useful error somewhere else?

I tried building Jena locally but it's failing for some reason. Upgraded to Java 11 and latest Maven, but still some annoying build error. Will figure out later what's wrong, but otherwise I would test to see what happens in this case, sorry.

Reply:
This isn't accessible from an assembler ATM.

Jetty has a test "if (maxThreads < minThreads)" but can change the use of min/max to make any negative numbers be "default".